lib/travis/build/script/go.rb

```diff
@@ -149,7 +150,7 @@ def comparable_go_version
           end
 
           def go_get_cmd
-            (go_version == 'go1' || comparable_go_version < Gem::Version.new('1.2')) ? 'go get' : 'go get -t'
+            (go_version == 'go1' || (go_version != 'tip' && comparable_go_version < Gem::Version.new('1.2'))) ? 'go get' : 'go get -t'
```

Comment:
On a second thought, this is not quite save us much, since the two place `#comparable_go_version` appears are not quite equal; one compares with `1.2`, the other with `1.5`.